renku/core/models/projects.py

```diff
@@ -88,8 +98,52 @@ def __attrs_post_init__(self):
                 # this assumes the project is being newly created
                 self.creator = Creator.from_git(self.client.repo)
 
-        if not self._id and self.client:
-            self._id = self.client.project_id
+        if not self._id or self._id.startswith('file://'):
+            self._id = self.project_id
+
+    @property
+    def project_id(self):
+        """Return the id for the project based on the repo origin remote."""
+        from renku.core.models.git import GitURL
+
+        # RENKU_DOMAIN environment variable is the host override
+        host = os.environ.get('RENKU_DOMAIN')
+        owner = self.creator.short_name if self.creator else 'NULL'
+        name = self.name
+
+        if self.client and not host:
+            remote_name = 'origin'
+            try:
+                remote_branch = \
+                    self.client.repo.head.reference.tracking_branch()
+                if remote_branch is not None:
+                    remote_name = remote_branch.remote_name
+            except TypeError:
+                pass
+
+            try:
+                url = GitURL.parse(self.client.repo.remotes[remote_name].url)
+
+                # Remove gitlab. unless running on gitlab.com.
+                hostname_parts = url.hostname.split('.')
+                if len(hostname_parts) > 2 and hostname_parts[0] == 'gitlab':
+                    hostname_parts = hostname_parts[1:]
+                url = attr.evolve(url, hostname='.'.join(hostname_parts))
```

Comment:
I think we can do a little bit better here. Improving a comment by explaining why and adding a test case would be a great addition to this. 

I would propose that we extract it to its own function and put proper docstring explaining what and why. That way swapping/editing this part for additional cases in the future becomes much easier.

Reply:
actually I did that already :)

Reply:
oh - did you mean just the gitlab part? I moved the whole piece to `core/management/repository.py`